### PR TITLE
Removing the use of eval in Node.js

### PR DESCRIPTION
**Pull request by itayperry**
#### Purpose

**Not allowing the use of eval in Node.js** - as decided in this discussion: https://github.com/sinonjs/fake-timers/issues/319#issuecomment-632794814

In this pull request using an `eval()` in `Node.js` env will throw an error.
At first, I've seen that there are 4 tests that test things who **do not** require `eval()` at all (thinking about https://github.com/sinonjs/fake-timers/pull/327#discussion_r429342307), I've replaced the strings in these tests with empty functions. 

In addition, I've added 2 `describe()` test blocks to test the use of eval (instead of the previous eval tests - who ignored the environments), once in Node.js and once in other environments. 

I've added a **global variable** that will only exist in a `Node.js` environment

### CODE REVIEW

**Comment by itayperry** [comment]
Hi, I rebased and the conflict that appeared at the beginning is now gone :)

**Comment by itayperry** [comment]
Cool :) I have one more thing to do regarding the `eval()` - to warn or show a console message when the eval is used (not in Node.js)

**Comment by benjamingr** [comment]
This isn't merged yet because it's semver-major

**Comment by fatso83** [comment]
@benjamingr Not sure why that is a reason preventing merging? Just bump the version and release, or are you waiting to have this automated?

**Comment by benjamingr** [comment]
@fatso83 I didn't know when a release would happen and didn't want to release a major without talking to anyone mostly

**Comment by fatso83** [comment]
Oh! That's very considerate, but a release happens when people have time to do it. If we need to coordinate something it's usually enough to tag the maintainer team. You and Simen are the most active these days on this project, so you are probably best placed to make a decision on when to do it. @mroderick is also more active than me, so I should not be a blocker. We split out these subprojects to let them be maintained by people with the most interest in them after all, so that we could focus on the main Sinon project.

Basically, it should not matter. Nothing breaks if you do a semver upgrade.

**Comment by mroderick** [comment]
@benjamingr if you share your username on the npm registry with me, then I can add you to the `sinonjs` organisation, and you can publish the package yourself

**Comment by benjamingr** [comment]
https://www.npmjs.com/~benjamin.gruenbaum @mroderick

**Comment by benjamingr** [comment]
It also might make sense from a security PoV to remove peer5 (my old employer from when I started contributing which I believe was added) unless they've made commits since.

**Comment by mroderick** [comment]
@benjamingr I've sent an invite to the username and have removed peer5

**Comment by itayperry** [comment]
Is this PR ok? Is there anything I need to fix or something?

**Comment by mroderick** [comment]
Something seems to have gone wrong with updating this from `master`,  as `isNearInfiniteLimit` is `undefined`. The tests fail because of out of heap memory.

File: src/fake-timers-src.js [hunk 1]
```diff
@@ -267,6 +267,22 @@ function withGlobal(_global) {
             throw new Error("Callback must be provided to timer calls");
         }
 
+        if (isNearInfiniteLimit) {
+            timer.error = new Error();
+        }
+
```
Comment (fatso83) [inline src/fake-timers-src.js:273]: This was added in https://github.com/sinonjs/fake-timers/commit/52d6dba2ede53fd0be88ff750f59937fbb10daee, but this flag has been gone for some time. Is undefined.

**Review by fatso83 — CHANGES_REQUESTED** [review]

File: src/fake-timers-src.js (hunk 1 above)
Comment (itayperry) [reply src/fake-timers-src.js:273]: Oh, I see, it is a part of the reverted merge that is now gone? Should I rebase again?

**Comment by itayperry** [comment]
I'm sorry for the mess, I'm having a hard time working with GitHub's commands and branches sometimes. 
@fatso83 would you like to try merging it again? The `isNearInfiniteLimit` is now gone and all tests pass as usual.

**Comment by fatso83** [comment]
This seems fine now, but as you can see from the status indicators on the individual commits, it still has a non-passing state. It is because of you not running Prettier. Just running `npm run  prettier:write` should fix you up 😄

**Comment by fatso83** [comment]
You  know what, I'll just fix it :)

**Comment by fatso83** [comment]
Thanks, this has now been merged.

**Comment by itayperry** [comment]
Awesome! Thank you! :)
